File: test/fixtures/java-unirest/put_xput.java
import kong.unirest.HttpResponse;
import kong.unirest.Unirest;

class Main {
    public static void main(String[] args) {
            HttpResponse<String> httpResponse
               = Unirest.put("http://localhost:28139/twitter/_mapping/user?pretty")
            .header("Content-Type", "application/json")
            .body("{\"properties\": {\"email\": {\"type\": \"keyword\"}}}")
            .asString();
    }
}

File: test/fixtures/kotlin-ktor/put_xput.kt
import io.ktor.client.HttpClient
import io.ktor.client.request.header
import io.ktor.client.request.put
import io.ktor.client.request.setBody
import io.ktor.client.statement.bodyAsText
import io.ktor.http.isSuccess
import kotlinx.coroutines.runBlocking

val client = HttpClient()

runBlocking {
    val response = client.put("http://localhost:28139/twitter/_mapping/user?pretty") {
        setBody("{\"properties\": {\"email\": {\"type\": \"keyword\"}}}")
        header("Content-Type", "application/json")
    }
    if (!response.status.isSuccess()) throw Exception("Request failed with status code ${response.status}")

    println(response.bodyAsText())
}
